mypy/build.py

```diff
@@ -539,6 +544,7 @@ def __init__(self, data_dir: str,
         self.version_id = version_id
         self.modules = {}  # type: Dict[str, MypyFile]
         self.missing_modules = set()  # type: Set[str]
+        self.module_discovery = module_discovery
         self.plugin = plugin
         self.semantic_analyzer = SemanticAnalyzerPass2(self.modules, self.missing_modules,
                                                   lib_path, self.errors, self.plugin)
```

Comment:
I'm not sure why BuildManager should get `lib_path` as a separate argument at all anymore; that should be encapsulated in `ModuleDiscovery`. Passing in both is just an opportunity for them to be out of sync, or for some code to use `lib_path` improperly, bypassing the `ModuleDiscovery` logic.

The only thing `BuildManager` seems to use `lib_path` for now is to pass it along to `SemanaticAnalyzerPass2`, but the latter doesn't seem to do anything with it either that I can find; it just stashes it on `self` and never looks at it again. So unless I miss something, I'd eliminate it from both.

Reply:
Yes, thanks, I will do that.